csharp/src/Apache.Arrow.Adbc/Extensions/IArrowArrayExtensions.cs

```diff
@@ -273,20 +299,22 @@ public static class IArrowArrayExtensions
         /// </summary>
         private static string SerializeToJson(StructArray structArray, int index)
         {
-            Dictionary<String, object?>? jsonDictionary = ParseStructArray(structArray, index);
+            ExpandoObject? obj = ParseStructArray(structArray, index);
```

Comment:
What's the advantage of `ExpandoObject` over `Dictionary`? Do we think that people will want to use these as `dynamic`?

(Also, this is arguably a breaking change though in this case it can be argued that the previous behavior was broken in that we didn't specify the "correct" type being returned.)

Reply:
I switched it back to Dictionary ... probably easier to acesss properties dynamically that way.